python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheckPartConnections.java

```diff
@@ -0,0 +1,127 @@
+    // aws_cdk.aws_ec2.Connections "allow from" methods call
+    checkFqn("aws_cdk.aws_ec2.Connections.allow_from", checkPeerAndPortSensitivity(OTHER, PORT_RANGE));
+    checkFqn("aws_cdk.aws_ec2.Connections.allow_from_any_ipv4", checkPortSensitivity(PORT_RANGE));
+    checkFqn("aws_cdk.aws_ec2.Connections.allow_default_port_from", UnrestrictedAdministrationCheckPartConnections::checkPeerAndDefaultPortInConstructorCall);
+    checkFqn("aws_cdk.aws_ec2.Connections.allow_default_port_from_any_ipv4", UnrestrictedAdministrationCheckPartConnections::checkDefaultPortInConstructorCall);
+
+    // SecurityGroup.add_ingress_rule
+    checkFqn("aws_cdk.aws_ec2.SecurityGroup.add_ingress_rule", checkPeerAndPortSensitivity("peer", "connection"));
+  }
+
+  private static void checkPeerAndDefaultPortInConstructorCall(SubscriptionContext ctx, CallExpression callExpression) {
+    getArgument(ctx, callExpression, OTHER, 0)
+      .filter(flow -> flow.hasExpression(IS_SENSITIVE_PEER))
+      .ifPresent(flow -> checkDefaultPortInConstructorCall(ctx, callExpression));
+  }
+
+  private static void checkDefaultPortInConstructorCall(SubscriptionContext ctx, CallExpression callExpression) {
+    Expression expression = callExpression.callee();
+    if (expression.is(Tree.Kind.QUALIFIED_EXPR)) {
+      expression = ((QualifiedExpression) expression).qualifier();
+    }
+
+    // trace back the creation of the object to check if a sensitive port was specified as default
+    CdkUtils.ExpressionFlow flowObj = CdkUtils.ExpressionFlow.build(ctx, expression);
+    flowObj.getExpression(isCallExpression().and(isFqn("aws_cdk.aws_ec2.Connections")))
+      .map(CallExpression.class::cast)
+      .flatMap(callExpr -> getArgument(ctx, callExpr, "default_port"))
+      .ifPresent(flow -> {
+        if (flow.hasExpression(IS_SENSITIVE_PORT)) {
+          ctx.addIssue(getMethodExpression(callExpression), MESSAGE_BAD_METHOD);
+        }
+      });
+  }
+
+  private static BiConsumer<SubscriptionContext, CallExpression> checkPeerAndPortSensitivity(String peerName, String portName) {
+    return (ctx, callExpression) -> {
+      if (getArgument(ctx, callExpression, peerName, 0).filter(flow -> flow.hasExpression(IS_SENSITIVE_PEER)).isPresent()
+       && getArgument(ctx, callExpression, portName, 1).filter(flow -> flow.hasExpression(IS_SENSITIVE_PORT)).isPresent()) {
+        getArgument(ctx, callExpression, peerName, 0).ifPresent(flow -> flow.addIssue(MESSAGE_BAD_PEER));
+      }
+    };
+  }
+
+  private static BiConsumer<SubscriptionContext, CallExpression> checkPortSensitivity(String portName) {
+    return (ctx, callExpression) ->
+      getArgument(ctx, callExpression, portName, 0).ifPresent(
+        flow -> {
+          if(flow.hasExpression(IS_SENSITIVE_PORT)) {
+            ctx.addIssue(getMethodExpression(callExpression), MESSAGE_BAD_METHOD);
+          }
+        });
+  }
+
+  private static Expression getMethodExpression(CallExpression callExpression) {
```

Comment:
If I understood correctly, this is to get the tree to use as primary location.
I would rename this method to something like "getMethodPrimaryLocation"